### PR TITLE
PR: Gogs issue #79

### PR DESCRIPTION
**Pull request by nelsonic**
Working on:
+ [x] deploying Gogs to Raspberry Pi running Ubuntu.
  + [x] Connecting `Gogs` running on the RPi via: [/gogs/connect-to-raspberrypi.md](https://github.com/dwyl/learn-devops/blob/gogs-issue-%2379/gogs/connect-to-raspberrypi.md)
    + [x] `SSH` (`Git`)
    + [x] `REST API` (`HTTP`)
+ [x] deploying to Fly.io https://fly.io/docs/app-guides/git-gogs-server/
+ [x] connecting to the `Gogs` server on Fly.io 🤞

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab I've created a separate repo: https://github.com/dwyl/gogs-server 
Which has all the steps to deploy `gogs` on Fly.io :shipit: 

I've linked to it from this PR. 
Hopefully everything is clear. 
If not, please LMK.
Thanks.

File: gogs/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+for 3 reasons: <br />
+a) Keeps it fully isolated on a separate machine. 
+Which simulates "production". <br />
+b) Allows us to test connecting to it over a network. i.e. real world tests.<br />
+c) Means we can run the instance long-term 
+and store actual (backup) data on it.<br />
+
+There are quite a few steps for setting this up,
+so we documented them in a separate doc.
+See: [`/install-gogs-raspberrypi.md`]()
```
Comment (SimonLab) [inline gogs/README.md:89]: Do we need to specify the link:
```suggestion
See: [`/install-gogs-raspberrypi.md`](./install-gogs-raspberrypi.md)
```
Or is Github clever enough to created from the name?

File: gogs/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Once we had setup the `Gogs` server on the `RPi`
+with `nginx` and `systemd` to auto-boot 
+whenever we restart the `RPi`,
+we tested connecting to `Gogs` on the `RPi`
+and documented the steps:
+[`/connect-to-raspberrypi.md`]()
```
Comment (SimonLab) [inline gogs/README.md:96]:
```suggestion
[`/connect-to-raspberrypi.md`](./connect-to-raspberrypi.md)
```

File: gogs/README.md (hunk 1 above)
Comment (nelsonic) [reply gogs/README.md:89]: nope. GitHub is definitely not that smart. a link without an `src` just goes to the current page. I hadn't updated these cause I didn't have the full URLs yet. but let's add this and see if it works once on the `main` branch. 💭

File: gogs/install-gogs-raspberrypi.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+> We are using Ubuntu because it's _ubiquitous_
+> and will help with both Continuous Integration 
+> and deployment to Cloud/VPS later
+> so getting it working on the Pi is a good first step. ✅
+
+
+### **`Postgres`**
+
+The default DB for **`Gogs`** 
+is be **`MySQL`**,
```
Comment (SimonLab) [inline gogs/install-gogs-raspberrypi.md:33]:
```suggestion
is **`MySQL`**,
```

File: gogs/install-gogs-raspberrypi.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+sudo su -
+mkdir /root/certs && cd /root/certs
+```
+
+Create the self-signed certificate:
+```sh
+openssl req -new -newkey rsa:4096 -x509 -sha256 -days 365 -nodes -out MyCertificate.crt -keyout MyKey.key
+```
+
+Accept all the defaults for the certificate creation
```
Comment (SimonLab) [inline gogs/install-gogs-raspberrypi.md:287]:
```suggestion
Accept all the defaults for the certificate creation,
```

**Review by SimonLab — COMMENTED** [review]
LGMT, just a couple of typos on `install-gogs-raspberrypi.md` but otherwise :+1:

**Comment by nelsonic** [comment]
@SimonLab all your fixes have been applied to the branch. Please merge if you're happy. Thanks! ☀️

**Review by SimonLab — APPROVED** [review]
Looks good thanks

**Comment by nelsonic** [comment]
Merci Simon. 🚀

**Pull request merged**